app/Http/Controllers/OverviewController.php

```diff
@@ -0,0 +1,111 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+
+use App\Models\UserPref;
+use App\Models\UserWidget;
+use App\Models\Dashboard;
+use App\Models\Widget;
+use Auth;
+use LibreNMS\Config;
+use Toastr;
+
+class OverviewController extends Controller
+{
+    public function index(Request $request)
+    {
+        $view = Config::get('front_page');
+
+        if (view()->exists("overview.custom.$view")) {
+            return view("overview.custom.$view");
+        } elseif (method_exists($this, $view)) {
+            return $this->{$view}($request);
+        }
+
+        return $this->default($request);
+    }
+
+    public function default(Request $request)
+    {
+        $user = Auth::user();
+        $dashboards = Dashboard::allAvailable($user)->with('user:user_id,username')->get()->keyBy('dashboard_id');
+
+        // Split dashboards into user owned or shared
+        list($user_dashboards, $shared_dashboards) = $dashboards->partition(function ($dashboard) use ($user) {
+            return $dashboard->user_id == $user->user_id;
+        });
+
+
+
+        if (!empty($request->dashboard) && isset($dashboards[$request->dashboard])) {
+            // specific dashboard
+            $dashboard = $dashboards[$request->dashboard];
+        } else {
+            $user_default_dash = (int)UserPref::getPref($user, 'dashboard');
+            $global_default = (int)Config::get('webui.default_dashboard_id');
+
+            // load user default
+            if (isset($dashboards[$user_default_dash])) {
+                $dashboard = $dashboards[$user_default_dash];
+            // load global default
+            } elseif (isset($dashboards[$global_default])) {
+                $dashboard = $dashboards[$global_default];
+            // load users first dashboard
+            } elseif (!empty($user_dashboards)) {
+                $dashboard = $user_dashboards->first();
+            }
+
+            // specific dashboard was requested, but doesn't exist
+            if (!empty($request->dashboard)) {
+                Toastr::error(
+                    "Dashboard <code>#$request->dashboard</code> does not exist! Loaded <code>
+                    ".htmlentities($dashboard->dashboard_name)."</code> instead.",
+                    "Requested Dashboard Not Found!"
+                );
+            }
+        }
+
+        if (!isset($dashboard)) {
+            $new_dash = new Dashboard();
+
+            $new_dash->dashboard_name = 'Default';
+            $new_dash->user_id = $user->user_id;
+            $new_dash->save();
+
+            $dashboard = $new_dash;
+
+            // TODO: Is this still needed?
+            UserWidget::where('user_id', $user->user_id)
+              ->where('dashboard_id', 0)
+              ->update(['dashboard_id' => $new_dash->dashboard_id]);
+        }
+
+        $data = $dashboard
```

Comment:
$data = $dashboard->widgets;  will give you all the widgets.  Using it as a function starts a query.

Reply:
Yeah but we need to make a query, `widgets()` is actually `userWidget`, and we need `widget` column from `Widgets` table.